web3/providers/rpc.py

```diff
@@ -24,6 +25,7 @@ class HTTPProvider(JSONBaseProvider):
     endpoint_uri = None
     _request_args = None
     _request_kwargs = None
+    _middlewares = (http_retry_request_middleware, 'http_retry_request')
```

Comment:
I would expect this to need to be something like: `[(http_retry_request_middleware, 'http_retry_request')]`, with the named entry pair as the first element of an iterable. This is an example of when the Mock testing isn't as effective as we might hope.

Reply:
Fixed this, ended up using the NamedElementStack data structure to be able to merge with the outer_middlewares.